clustpy/deep/dcn.py

```diff
@@ -322,49 +321,40 @@ def fit(self, autoencoder: torch.nn.Module, trainloader: torch.utils.data.DataLo
             this instance of the _DCN_Module
         """
         # DCN training loop
-        # Init for count from original DCN code (not reported in Paper)
-        # This means centroid learning rate at the beginning is scaled by a hundred
-        count = torch.ones(self.centers.shape[0], dtype=torch.int32) * 100
         for _ in range(n_epochs):
             # Update Network
             for batch in trainloader:
-                loss = self._loss(batch, autoencoder, loss_fn, degree_of_space_preservation, degree_of_space_distortion,
+                loss = self._loss(batch, autoencoder, loss_fn, reconstruction_loss_weight, clustering_loss_weight,
                                   device)
                 # Backward pass - update weights
                 optimizer.zero_grad()
                 loss.backward()
                 optimizer.step()
-            # Update Assignments and Centroids
-            with torch.no_grad():
-                for batch in trainloader:
+                # Update Assignments and Centroids
+                with torch.no_grad():
```

Comment:
we execute minibatch kmeans for each batch. Before we executed it after each epoch. Corresponds to original implementation